### PR TITLE
feat: add cases of facility to frontend

### LINKED ISSUE #250: Add cases of facility to Frontend
## User story

I want to be able to see the amount of cases of a facility and not just a randomly generated number

## Acceptance criteria
- [ ] adjust the amount of open cases of a facility to actually be the amount of open cases from the backend(detail page, overview page)
- [ ] show the first 3 open cases linked to this facility and if you click it you are redirected to that specific case (if there are no or 1/2 of course show "no cases" or the 1/2 links)
- [ ] if there are more than 3 cases show "Go to all Cases" and link to the cases overview
- [ ] if there are exactly 4 cases just show the 4th case instead of "Go to all Cases"

## Definition of done (DoD)

-     Added only after week 5
-     The same for all features
-     Here goes the project specific part

## DoD general criteria

-     Feature has been fully implemented
-     Feature has been merged into the mainline
-     All acceptance criteria were met
-     Product owner approved features
-     All tests are passing
-     Developers agreed to release

### PR DESCRIPTION
**Pull request by Sungonogi**
I also changed the routing of the goBack Button to use location.back(). I first thought just cutting the current url until the / would be the best but location.back() is actually better.

It goes back to the last page so now if you click on view 3rd case and go back you are redirected back to the details page where you came from

Signed-off-by: Ingo Sternberg <ingo.sternberg@fau.de>
#250

### CODE REVIEW

File: libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts
Comment (HaruspexSan) [inline libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts:1]: what is this for? empty params?

File: libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html
Comment (HaruspexSan) [inline libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html:1]: What is the location.back() function?

**Review by Hydraneut — APPROVED** [review]

**Review by HaruspexSan — COMMENTED** [review]
looks good - besides the two comments on the files, why is it called: "facility.cases.length" and not smth. like facility.cases.count as it is not the length of a case but i assume the number of cases for a given facility. This is nitpicking, i know.

File: libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html
Comment (Sungonogi) [reply libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html:1]: I wrote it in the description of this PR

File: libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts
Comment (Sungonogi) [reply libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts:1]: The test did not work for some reason so I added it to fix the test

**Comment by Sungonogi** [comment]
It is .length because cases is an array and to get the length of it you do .length

File: libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts
Comment (HaruspexSan) [reply libs/cases/frontend/view/src/lib/components/create-case/create-case.component.spec.ts:1]: ok, got it. Sometimes these tests can be pesky

File: libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html
Comment (HaruspexSan) [reply libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.html:1]: welp, guess who didnt read it my bad

**Review by HaruspexSan — APPROVED** [review]
looks good - tank you for the quick answer

**Pull request merged**